### PR TITLE
autoenv: init at 0-unstable-2024-10-16

### PR DESCRIPTION
**Pull request by clebs**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by clebs** [comment]
1 package built:
autoenv

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349058`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>autoenv</li>
  </ul>
</details>

### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>autoenv</li>
  </ul>
</details>

### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>autoenv</li>
  </ul>
</details>

**Review by husjon — APPROVED** [review]

File: pkgs/by-name/au/autoenv/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    hash = "sha256-vZrsMPhuu+xPVAww04nKyoOl7k0upvpIaxeMrCikDio=";
+  };
+
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/share
+    cp -R $src $out/share/autoenv
+    chmod +x $out/share/autoenv
+  '';
```
Comment (FliegendeWurst) [inline pkgs/by-name/au/autoenv/package.nix:24]:
```suggestion
  installPhase = ''
    runHook preInstall
    mkdir -p $out/share
    cp -R $src $out/share/autoenv
    chmod +x $out/share/autoenv
    runHook postInstall
  '';
```

File: pkgs/by-name/au/autoenv/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  lib,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "autoenv";
+  version = "unstable-2024-10-16";
```
Comment (FliegendeWurst) [inline pkgs/by-name/au/autoenv/package.nix:9]:
```suggestion
  version = "0-unstable-2024-10-16";
```

File: pkgs/by-name/au/autoenv/package.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/share
+    cp -R $src $out/share/autoenv
+    chmod +x $out/share/autoenv
+  '';
+
+  meta = {
+    description = "Magic per-project shell environments.";
```
Comment (FliegendeWurst) [inline pkgs/by-name/au/autoenv/package.nix:27]:
```suggestion
    description = "Per-directory shell environments sourced from .env file";
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/au/autoenv/package.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    cp -R $src $out/share/autoenv
+    chmod +x $out/share/autoenv
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Per-directory shell environments sourced from .env file";
+    homepage = "https://github.com/hyperupcall/autoenv";
+    license = lib.licenses.mit;
+    mainProgram = "autoenv";
```
Comment (FliegendeWurst) [inline pkgs/by-name/au/autoenv/package.nix:32]: There is no binary at $out/bin/autoenv. You can create a symlink to $out/share/activate.sh in installPhase.

Comment (clebs) [reply pkgs/by-name/au/autoenv/package.nix:32]: That is what the module activation already does.

Comment (FliegendeWurst) [reply pkgs/by-name/au/autoenv/package.nix:32]: If you set `mainProgram = x`, there must be a file at `$out/bin/x`. I would suggest link $out/bin/autoenv to $out/share/activate.sh.

Comment (ambroisie) [reply pkgs/by-name/au/autoenv/package.nix:32]: I think `mainProgram` doesn't make sense for this package. Unlike e.g: `direnv`, there is no executable for autoenv, is there?

Comment (FliegendeWurst) [reply pkgs/by-name/au/autoenv/package.nix:32]: Oh right. Doesn't really make sense to execute the activation script like that. In that case you can just omit mainProgram from meta.

Comment (clebs) [reply pkgs/by-name/au/autoenv/package.nix:32]: I understand, will just remove that line then!

Comment (clebs) [reply pkgs/by-name/au/autoenv/package.nix]: Applied all suggestions and squashed the commits.

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**